mmv1/third_party/terraform/services/storage/resource_storage_bucket.go.tmpl

```diff
@@ -567,6 +567,86 @@ func ResourceStorageBucket() *schema.Resource {
 					},
 				},
 			},
+{{- if ne $.TargetVersionName "ga" }}
+			"ip_filter": {
+				Type:             schema.TypeList,
+				MaxItems:         1,
+				Optional:         true,
+				Description:      `The bucket IP filtering configuration.`,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"mode": {
+							Type: schema.TypeString,
+							Required: true,
+							Description: `The mode of the IP filter. Valid values are 'Enabled' and 'Disabled'.`,
+							ValidateFunc: validation.StringInSlice([]string{"Enabled", "Disabled"}, false),
+						},
+						"public_network_source": {
+							Type:        schema.TypeList,
+							MaxItems:    1,
+							Optional:    true,
+							Description: `The bucket IP filtering configuration.`,
+							Elem:        &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"allowed_ip_cidr_ranges": {
+										Type:        schema.TypeList,
+										Required:    true,
+										Description: "The list of public IPv4, IPv6 cidr ranges that are allowed to access the bucket.",
+										Elem:        &schema.Schema{
+											Type: schema.TypeString,
+											ValidateFunc: validation.IsCIDR,
+										},
+									},
+								},
+							},
+						},
+						"vpc_network_sources": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							Description: `The bucket IP filtering configuration.`,
+							Elem:        &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"allowed_ip_cidr_ranges": {
+										Type:     schema.TypeList,
+										Required: true,
+										Description: "The list of public IPv4, IPv6 cidr ranges that are allowed to access the bucket.",
+										Elem:     &schema.Schema{
+											Type: schema.TypeString,
+											ValidateFunc: validation.IsCIDR,
+										},
+									},
+									"network": {
+										Type:        schema.TypeString,
+										Required:    true,
+										Description: "Name of the network. Format: projects/{PROJECT_ID}/global/networks/{NETWORK_NAME}",
+										ValidateFunc: validation.StringMatch(regexp.MustCompile("projects/.+/global/networks/.+"), ""),
```

Comment:
FWIW, networks are passed into a variety of resources across the provider and I don't see them being validated with a regex like this. I'm not sure of the reason, but it could be to support other formats or future changes (like `https://www.googleapis.com/compute/v1/` prefix or similar). If you're sure you want this, I think it's ok to keep, but I would lean toward removing it.